applications/ShapeOptimizationApplication/python_scripts/algorithm_steepest_descent.py

```diff
@@ -142,34 +157,83 @@ def __analyzeShape(self):
 
         self.model_part_controller.DampNodalVariableIfSpecified(DF1DX)
 
+    # --------------------------------------------------------------------------
+    def __adjustStepSize(self):
+        current_a = self.step_size
+
+        # Compare actual and estimated improvement using linear information
+        dfda1 = 0.0
+        for node in self.design_surface.Nodes:
+            s1 = node.GetSolutionStepValue(SEARCH_DIRECTION)
+            dfds1 = node.GetSolutionStepValue(DF1DX_MAPPED)
```

Comment:
maybe add a comment that makes clear that `DF1DX_MAPPED` still has the values from the previous step.